src/aind_registration_evaluation/util/extract_roi.py

```diff
@@ -0,0 +1,291 @@
+        patches.append((z_min, y_min, x_min, z_max, y_max, x_max))
+    
+    merged_patches = []
+    while patches:
+        z_min, y_min, x_min, z_max, y_max, x_max = patches.pop(0)
+        merged = False
+        for i, (mz_min, my_min, mx_min, mz_max, my_max, mx_max) in enumerate(merged_patches):
+            if not (z_max < mz_min or z_min > mz_max or y_max < my_min or y_min > my_max or x_max < mx_min or x_min > mx_max):
+                overlap_patch = (max(z_min, mz_min), max(y_min, my_min), max(x_min, mx_min), min(z_max, mz_max), min(y_max, my_max), min(x_max, mx_max))
+                max_patch = (min(z_min, mz_min), min(y_min, my_min), min(x_min, mx_min), max(z_max, mz_max), max(y_max, my_max), max(x_max, mx_max))
+                volume_overlap = (overlap_patch[3] - overlap_patch[0]) * (overlap_patch[4] - overlap_patch[1]) * (overlap_patch[5] - overlap_patch[2])
+                volume_max_patch = (max_patch[3] - max_patch[0]) * (max_patch[4] - max_patch[1]) * (max_patch[5] - max_patch[2])
+                if volume_overlap/volume_max_patch > overlapThreshold:
+                    merged_patches[i] = max_patch
+                    if i<(len(merged_patches)-1):
+                        patches += merged_patches[i+1:]
+                        merged_patches = merged_patches[:i+1]
+                    merged = True
+                    break
+        if not merged:
+            merged_patches.append((z_min, y_min, x_min, z_max, y_max, x_max))
+    return merged_patches
+
+def get_ROIs(img1, img2, img1_binaryThreshold, img2_binaryThreshold, maxCentroidDistance, overlapThreshold):
+    '''
+        Calls the functions above and returns a dictionary of {patch_coordinates: patch_volume}
+
+        Parameters
+        ------------------------
+        img1, img2: numpy array of both images. Must be 3d of the same shape
+
+        img1_binaryThreshold, img2_binaryThreshold: threshold to create a binary image based on intensity (int)
+        
+        maxCentroidDistance: pixel count for allowed distance of centroids to be seen as a match
+        
+        overlapThreshold: volume overlap theshold above which two patches are merged to their max coordinates (float)
+
+        Returns
+        ------------------------
+        dictionary of {patch_coordinates: patch_volume}
+    '''
+    img1_props = get_props(img1, img1_binaryThreshold)
+    img2_props = get_props(img2, img2_binaryThreshold)
+    print("ConfocalProps:", len(img1_props), "CorticalProps:", len(img2_props))
+    distances = get_centroid_distances(img1_props, img2_props, maxCentroidDistance)
+    print("Cell Matches:", len(distances))
+    patches = create_patches(distances, img1.shape, overlapThreshold)
+    patch_dict = {r: ((r[3]-r[0]) * (r[4]-r[1]) * (r[5]-r[2])) for r in patches}
+    return patch_dict
+
+def get_ROIs_cellpose(img1, img2, maxCentroidDistance, overlapThreshold):
```

Comment:
this function and the one above seem pretty similar, please, optimize if possible

Reply:
A helper function could be useful here